src/client.rs

```diff
@@ -162,6 +213,67 @@ impl Client {
     pub async fn leases(&mut self) -> Result<LeaseLeasesResponse> {
         self.lease.leases().await
     }
+
+    /// Lock acquires a distributed shared lock on a given named lock.
+    /// On success, it will return a unique key that exists so long as the
+    /// lock is held by the caller. This key can be used in conjunction with
+    /// transactions to safely ensure updates to etcd only occur while holding
+    /// lock ownership. The lock is held until Unlock is called on the key or the
+    /// lease associate with the owner expires.
+    #[inline]
+    pub async fn lock(
+        &mut self,
+        name: impl Into<Vec<u8>>,
+        options: Option<LockOptions>,
+    ) -> Result<LockResponse> {
+        self.lock.lock(name, options).await
+    }
+
+    /// Unlock takes a key returned by Lock and releases the hold on lock. The
+    /// next Lock caller waiting for the lock will then be woken up and given
+    /// ownership of the lock.
+    #[inline]
+    pub async fn unlock(&mut self, key: impl Into<Vec<u8>>) -> Result<UnlockResponse> {
+        self.lock.unlock(key).await
+    }
+
+    /// auth_enable enables authentication.
+    #[inline]
+    pub async fn auth_enable(&mut self) -> Result<AuthEnableResponse> {
+        self.auth.auth_enable().await
+    }
+
+    /// auth_disable disables authentication.
+    #[inline]
+    pub async fn auth_disable(&mut self) -> Result<AuthDisableResponse> {
+        self.auth.auth_disable().await
+    }
+}
+
+/// tuple user contains the value of (name, password)
+#[derive(Debug, Default, Clone)]
+pub struct UserInfo(String, String);
+
+/// Options for `Connect` operation.
+#[derive(Debug, Default, Clone)]
+#[repr(transparent)]
+pub struct ConnectOptions {
+    user: Option<UserInfo>,
+}
+
+impl ConnectOptions {
+    /// name is the identifier for the distributed shared lock to be acquired.
+    #[inline]
+    pub fn with_user(mut self, name: String, password: String) -> Self {
```

Comment:
Declare `name` and `password` as `impl Into<String>` is better.

Reply:
OK